### DISCUSSION TITLE
Moving data to new greater capacity drive for expanding storage space

### DISCUSSION THREAD

**Discussion by creeew** [minio/minio#15522] [Q&A] [answered]
I'm looking for a easy way to expand a tiny distributed MinIO server which have 4 node and every node have one drive.
Expanding by server pool need too much resources for me. I'm trying replace the node's drive to expand storage space.
Thanks for the healing mechanism, replaced new greater capacity drive can recover the data automatically. Once every drives in cluster upgraded the whole cluster's storage space expanded.

But healing really cost too much time for me. Stopping the whole cluster when upgrading drive is acceptable to me, I move data to new drive on backend when cluster stopped. And everything looks good when cluster restarted.

According to the document: https://docs.min.io/minio/baremetal/installation/restore-minio.html#id2
Migrating mutually looks like a bad approach, could you tell me whether it is correct that migrate data to new drive mutually when all nodes down?

**Comment by harshavardhana** [comment, marked as answer]
You need to just setup new cluster and mc mirror the content or setup two pooled setup decom the older pool is to newer more denser drives.

**Reply by creeew** [reply]
Thanks for your answer.
Looks like we got no chance to spanding drivers by replacing the larger drive. mc mirror the content to new cluster or decom older pool to newer pool need more resource  and more step.
Would you be so kind to tell me more about the disadvantage of replacing larger drive and move data mutually when whole cluster down?
btw, I found a error log when I get file after move old data to new drive
`Error: Disk: /mnt/minio/data1  -> vvversioning/123.zip/43bdf31f-21de-44f4-9f40-ee0ba9a56ef7/part.1 - content hash does not match - expected 0000000000000000000000000000000000000000000000000000000000000000, got 3d7b653910b5153f170313059d765b6c6fbca47eeb174f76f3a561ee44dd4576 (*errors.errorString)
       3: internal/logger/logger.go:259:logger.LogIf()
       2: cmd/bitrot-streaming.go:183:cmd.(*streamingBitrotReader).ReadAt()
       1: cmd/erasure-decode.go:165:cmd.(*parallelReader).Read.func1()
`

**Reply by harshavardhana** [reply]
It will not work @creeew - that is too complicated

**Reply by creeew** [reply]
Alright, thanks anyway
